docs/user-guide/templates.md

```diff
@@ -0,0 +1,64 @@
+# Templates
+
+Templates are snippets of predefined code that people can use to replace a job definition in a [screwdriver.yaml](./configuration/index.md).
+
+## Job Templates
+
+A job template contains a series of predefined steps along with a selected Docker image.
+
+### Using a job template
+
+To use a job template, define a `screwdriver.yaml`:
+
+```
+jobs:
+   main:
+      template: template_name@1.3
+```
+
+Screwdriver takes the template configuration and plugs it in, so that the `screwdriver.yaml` becomes:
+
+```
+jobs:
+   main:
+        image: node:6
+        steps:
+          - install: npm install
+          - test: npm test
+          - echo: echo $FOO
+        environment:
+           FOO: bar
+        secrets:
+          - NPM_TOKEN
+```
+
+### Creating a job template
+
+To create a job template, create a new repo with a `screwdriver-template.yaml` file. The file should contain a name, version, description, maintainer email, and a config with an image and steps.
+
+Example `screwdriver-template.yaml`:
+
+```
+name: test
+version: 1.3
+description: template for testing
+maintainer: foo@bar.com
+config:
+  image: node:6
+  steps:
+    - install: npm install
+    - test: npm test
+    - echo: echo $FOO
+  environment:
+    FOO: bar
+  secrets:
+     - NPM_TOKEN
+```
+
+Create a Screwdriver pipeline with your template repo and run the build to validate and publish it.
```

Comment:
Should probably mention how to publish and validate
https://github.com/screwdriver-cd/template-main#publishing-a-template